lib/src/config/dictionaries.rs

```diff
@@ -0,0 +1,193 @@
+                        let json: serde_json::Value =
+                            serde_json::from_str(&data).map_err(|_| DictionaryConfigError::DictionaryFileWrongFormat {
+                                name: name.to_string(),
+                            })?;
+                        let dict = json.as_object().ok_or_else(|| {
+                            DictionaryConfigError::DictionaryFileWrongFormat {
+                                name: name.to_string(),
+                            }
+                        })?;
+                        if dict.len() > DICTIONARY_MAX_LEN {
+                            return Err(DictionaryConfigError::DictionaryCountTooLong {
+                                name: name.to_string(),
+                                size: DICTIONARY_MAX_LEN.try_into().unwrap(),
+                            });
+                        }
+
+                        event!(
+                            Level::INFO,
+                            "checking if the items in dictionary '{}' adhere to Fastly's API",
+                            name
+                        );
+                        for (key, value) in dict.iter() {
+                            if key.chars().count() > DICTIONARY_ITEM_KEY_MAX_LEN {
+                                return Err(DictionaryConfigError::DictionaryItemKeyTooLong {
+                                    name: name.to_string(),
+                                    key: key.clone(),
+                                    size: DICTIONARY_ITEM_KEY_MAX_LEN.try_into().unwrap(),
+                                });
+                            }
+                            let value = value
+                                .as_str()
+                                .ok_or_else(|| {
+                                    DictionaryConfigError::DictionaryItemValueWrongFormat {
+                                        name: name.to_string(),
+                                        key: key.clone(),
+                                    }
+                                })?;
+                            if value.chars().count() > DICTIONARY_ITEM_VALUE_MAX_LEN {
+                                return Err(DictionaryConfigError::DictionaryItemValueTooLong {
+                                    name: name.to_string(),
+                                    key: key.clone(),
+                                    size: DICTIONARY_ITEM_VALUE_MAX_LEN.try_into().unwrap(),
+                                });
+                            }
+                        }
+                        Ok(Dictionary {
+                            name: name.clone().try_into()?,
+                            file,
+                        })
+                    })
+                    .map_err(|err| FastlyConfigError::InvalidDictionaryDefinition {
+                        name: name.clone(),
+                        err,
+                    })
+                    .map(|def| (def.name.clone(), def))
```

Comment:
```suggestion
                        let name = name.parse()?;
                        Ok((name, Dictionary { file }))
                    })
                    .map_err(|err| FastlyConfigError::InvalidDictionaryDefinition {
                        name: name.clone(),
                        err,
                    })
```

Reply:
(_see the comments attached to `struct Dictionary`, as well as the `DictionaryName: TryFrom<String>` implementation_)